pkg/translator/prometheusremotewrite/helper_v2.go

```diff
@@ -0,0 +1,108 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package prometheusremotewrite // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/prometheusremotewrite"
+
+import (
+	"fmt"
+	"log"
+	"slices"
+
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/model/labels"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	conventions "go.opentelemetry.io/collector/semconv/v1.25.0"
+
+	prometheustranslator "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/prometheus"
+)
+
+// createAttributes creates a slice of Prometheus Labels with OTLP attributes and pairs of string values.
+// Unpaired string values are ignored. String pairs overwrite OTLP labels if collisions happen and
+// if logOnOverwrite is true, the overwrite is logged. Resulting label names are sanitized.
+func createAttributesV2(resource pcommon.Resource, attributes pcommon.Map, externalLabels map[string]string,
```

Comment:
This method looks almost identical to `createAttributes`, the only difference is what's been returned. All the common heavy lifting here seems to be sorting strings and verifying the presence of `service.name` and `service.name.id`. It would be wonderful if we could avoid code duplications here by moving similar things into a separate function that returns an array of strings.

`createAttributes` and `createAttributesv2` would both call these new function and then just transform the array of strings into `[]prompb.Label` and `label.Labels` respectively

Reply:
Addressed based on work by @aknuds1. thank you.